sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/TableItemAccessHelper.java

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.tables.implementation;
+
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.data.tables.implementation.models.TableResponseProperties;
+import com.azure.data.tables.models.TableItem;
+
+import java.util.function.Function;
+
+/**
+ * Helper class to access internal values of {@link TableItem}.
+ */
+public final class TableItemAccessHelper {
+    private static Function<TableResponseProperties, TableItem> creator;
+
+    /**
+     * Sets the {@link TableItem} creator.
+     *
+     * @param creator The {@link TableItem} creator.
+     */
+    public static void setTableItemCreator(Function<TableResponseProperties, TableItem> creator) {
+        TableItemAccessHelper.creator = creator;
+    }
+
+    /**
+     * Creates a {@link TableItem}.
+     *
+     * @param properties The {@link TableResponseProperties} used to construct the table.
+     * @return The created {@link TableItem}.
+     */
+    public static TableItem createItem(TableResponseProperties properties) {
+        if (creator == null) {
+            // Since the access helper needs to access a constructor the class needs to be initialized first.
+            try {
+                Class.forName(TableItem.class.getName(), true, TableItemAccessHelper.class.getClassLoader());
+            } catch (ReflectiveOperationException ex) {
+                throw new ClientLogger(TableItemAccessHelper.class).logExceptionAsError(
```

Comment:
can we use static logger instead?

Reply:
I'd rather stick with this as if this path is reached this will likely end the process, or at least result in whatever happens next to not work, and this defers the creation of `ClientLogger` to when it is needed instead of eagerly being initialized in the clinit.